src/shogun/base/class_list.h

```diff
@@ -17,16 +17,34 @@
 
 #include <set>
 #include <string>
+#include <queue>
 
 namespace shogun {
 	class SGObject;
 
+	template<typename classType>
+	std::string_view name_lookup()
+	{
+		if constexpr(std::is_same_v<classType, Machine>)
+		{
+			return "Machine";
+		}
+		else 
+		{
+			return "SGObject";
+		}
+	};
```

Comment:
I think this should be in base_types.h and also have the same base types that are forward declared there